controllers/validate.go

```diff
@@ -0,0 +1,49 @@
+package controllers
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"k8s.io/klog/v2"
+)
+	
+// DockerValidate is POST by us for validate the process.
+// The following parameters are recognized in callback data:
+// state (required): Accepted values are success, failure, and error. If the state isn’t success,
+// the Webhook chain is interrupted. description: A string containing miscellaneous information 
+// that is available on Docker Hub. Maximum 255 characters. context: A string containing the 
+// context of the operation. Can be retrieved from the Docker Hub. Maximum 100 characters.
+// target_url: The URL where the results of the operation can be found. Can be retrieved on the 
+// Docker Hub.
+type DockerValidate struct {
+	State       string `json:"state"`
+	Description string `json:"description"`
+	Context     string `json:"context"`
+	TargetURL   string `json:"target_url"`
+}
+
+// Validate send a POST request to callback_url containing a valid JSON body.
+func (v *DockerValidate) Validate(c string) (res map[string]interface{}, err error) {
```

Comment:
Two things here, the first one is this `c string` argument. Is this an URL? If yes then we should use an `url.URL` argument here instead (makes everything more clear). The second thing is this return value, you could create a proper type for it instead of `map[string]interface{}` that does not say much.

Reply:
"c string" is a URL string. We cannot use url.URL in argument to http.Postform.